### PR TITLE
fix(ci): only run the full sync test on mergify queue PRs

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

We want to run the full sync test exactly once before merging each PR.

### GitHub API Reference

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-based-on-the-head-or-base-branch-of-a-pull-request-1

## Solution

- only run the full sync test on `mergify/merge-queue/` branches

## Review

@gustavovalverde or @dconnolly can review this PR.

### Reviewer Checklist

  - [x] The full sync test doesn't run in this PR
  - [x] The full sync test runs in the Mergify queue PR for this branch

## Follow Up Work

- make the full sync test faster
- require the full sync test before merging Mergify queue PRs

### CODE REVIEW

File: .github/workflows/test-full-sync.yml
Comment (gustavovalverde) [inline .github/workflows/test-full-sync.yml]:
```suggestion
```

Having the comment here might be confusing. Just inside the job should be enough to understand the `if` statement

Comment (gustavovalverde) [inline .github/workflows/test-full-sync.yml:9]:
```suggestion
```

Having the comment here might be confusing. Just inside the job should be enough to understand the `if` statement

**Review by gustavovalverde — CHANGES_REQUESTED** [review]
Shouldn't we consider adding `Test full Mainnet sync` job to Mergify's configuration file as a required check?

**Comment by dconnolly** [comment]
> Shouldn't we consider adding `Test full Mainnet sync` job to Mergify's configuration file as a required check?

Once upon a time we said we wanted the full sync test to always run, but not block merge, because it takes a long time. Is that still what we want?

File: .github/workflows/test-full-sync.yml
Comment (teor2345) [reply .github/workflows/test-full-sync.yml:9]: Our priority is getting CI stable and working, so I'd like to focus on this specific bug fix in this PR.

These questions are about functionality changes. Can you open discussions or tickets for these questions?

Once we have fixed this bug, and the test is stable, we'll have better information to make these decisions. Then we can schedule those changes in a sprint. (I'd like to check with the team before we make changes with a big impact on CI performance or failure rates.)

> > Shouldn't we consider adding `Test full Mainnet sync` job to Mergify's configuration file as a required check?

We can't make this change right now, because the job is currently failing. It took more than 6 hours on the most recent PR.

> Once upon a time we said we wanted the full sync test to always run, but not block merge, because it takes a long time. Is that still what we want?

That is the current configuration, and it will continue to be the configuration after this PR merges.

I think our decision will depend on how fast and reliable the full sync test is.

**Comment by teor2345** [comment]
Our priority is getting CI stable and working, so I'd like to focus on this specific bug fix in this PR.

These questions are about functionality changes. Can you open discussions or tickets for these questions?

Once we have fixed this bug, and the test is stable, we'll have better information to make these decisions. Then we can schedule those changes in a sprint. (I'd like to check with the team before we make changes with a big impact on CI performance or failure rates.)

> > Shouldn't we consider adding `Test full Mainnet sync` job to Mergify's configuration file as a required check?

We can't make this change right now, because the job is currently failing. It took more than 6 hours on the most recent run in PR #3717:
https://github.com/ZcashFoundation/zebra/runs/5450677003?check_suite_focus=true

> Once upon a time we said we wanted the full sync test to always run, but not block merge, because it takes a long time. Is that still what we want?

That is the current configuration, and it will continue to be the configuration after this PR merges.

I think our final decision will depend on how fast and reliable the full sync test is.

File: .github/workflows/test-full-sync.yml
Comment (teor2345) [reply .github/workflows/test-full-sync.yml]: I was confused about how often this workflow run, because the full filters aren't documented at the top of the file.

But this fix is urgent, so I'm happy to remove the extra docs if you want.

**Review by gustavovalverde — APPROVED** [review]
LGTM

**Comment by dconnolly** [comment]
> - [x] The full sync test runs in the Mergify queue PR for this branch

https://github.com/ZcashFoundation/zebra/runs/5458521857?check_suite_focus=true

**Pull request merged**
